project/src/components/card/card.tsx

```diff
@@ -1,24 +1,43 @@
-function Card(): JSX.Element {
+import { Link } from 'react-router-dom';
+import { AppRoute } from '../../const';
+import { Offer } from '../../types/offer';
+import { calculateRatingWidth, upperFirstLetter } from '../../utils';
+
+type CardProps = {
+  offer: Offer;
+  onCardHover: (id: number | null) => void;
+}
+
+function Card({offer, onCardHover}: CardProps): JSX.Element {
+
+
   return (
-    <article className="cities__card place-card">
+    <article
+      className="cities__card place-card"
+      onMouseEnter={() => onCardHover(offer.id)}
+      onMouseLeave={() => onCardHover(null)}
+    >
+
+      { offer.isPremium &&
       <div className="place-card__mark">
         <span>Premium</span>
```

Comment:
Лучше использовать круглые скобки
```suggestion
      {offer.isPremium && (
        <div className="place-card__mark">
          <span>Premium</span>
          ....
        )}
```